tests/plotting/test_plot_manager.py

```diff
@@ -606,3 +606,41 @@ def raising_callback(event: PlotRefreshEvents, _: PlotRefreshResult):
     expected_result.processed = len(env.dir_1) + len(env.dir_2)
     expected_result.remaining = 0
     await env.refresh_tester.run(expected_result)
+
+
+@pytest.mark.asyncio
+async def test_recursive_plot_scan(test_plot_environment: TestEnvironment) -> None:
+    env: TestEnvironment = test_plot_environment
+    # Create a directory tree with some subdirectories containing plots, others not.
+    root_plot_dir = env.root_path / "root"
+    sub_dir_0: TestDirectory = TestDirectory(root_plot_dir / "0", env.dir_1.plots[0:2])
+    sub_dir_0_1: TestDirectory = TestDirectory(sub_dir_0.path / "1", env.dir_1.plots[2:3])
+    sub_dir_1: TestDirectory = TestDirectory(root_plot_dir / "1", [])
+    sub_dir_1_0: TestDirectory = TestDirectory(sub_dir_1.path / "0", [])
+    sub_dir_1_0_1: TestDirectory = TestDirectory(sub_dir_1_0.path / "1", env.dir_1.plots[3:7])
+
+    # List of all the plots in the directory tree
+    expected_plot_list = sub_dir_0.plot_info_list() + sub_dir_0_1.plot_info_list() + sub_dir_1_0_1.plot_info_list()
+
+    # Adding the root without `recursive_plot_scan` and running a test should not load any plots (match an empty result)
+    expected_result = PlotRefreshResult()
+    add_plot_directory(env.root_path, str(root_plot_dir))
+    await env.refresh_tester.run(expected_result)
+
+    # Set the recursive scan flag in the config
+    with lock_and_load_config(env.root_path, "config.yaml") as config:
+        config["harvester"]["recursive_plot_scan"] = True
+        save_config(env.root_path, "config.yaml", config)
+
+    # With the flag enabled it should load all expected plots
+    expected_result.loaded = expected_plot_list  # type: ignore[assignment]
+    expected_result.removed = []
+    expected_result.processed = len(expected_plot_list)
+    expected_result.remaining = 0
+    await env.refresh_tester.run(expected_result)
+
+    # Adding the subdirectories also should not lead to some failure or duplicated loading
```

Comment:
The plots are loaded into a dict based on the configuration path entry.  Do we care which one the plots show up under?  I feel like they should, maybe, rather be in the most specific path they match but I'm not sure it really matters in terms of actual usage.

Reply:
Doesn't matter, see https://github.com/Chia-Network/chia-blockchain/blob/main/chia/plotting/manager.py#L140-L141.